### ISSUE TITLE
Validation error when trying to save a buffer to the database

### ISSUE THREAD

**Issue by mattcodez** [balderdashy/waterline#1200]
I'm experiencing this same issue referenced in #792 that was fixed in 0.10.16, but I'm running 0.10.26. My code
```
setProperties: async function({siteTitle = 'Continuum', logo}){
    let cc = await ClientConfig.findOrCreate({}, {siteTitle});
    if (logo){
      //let image = await fs.readFileAsync(logo.fd);
      let image = fs.readFileSync(logo.fd);
      console.log(image);
      cc.logo = image;
    }
    return cc.save();
  }
```
The error
```
Invalid attributes sent to ClientConfig:
 • logo
   • `undefined` should be a binary (instead of "[object Object]", which is a object)
```
The log shows buffer data above the error so that seems to be coming in just fine, not sure what to look at beyond this.

**Comment by mattcodez** [comment]
Switched to `sails-disk` to see if this was specific to `sails-mongo` but the errors persist.

**Comment by devinivy** [comment]
This is likely due to some deep cloning.  @tjwebb and I are on the warpath to remove unnecessary clones from the library :)  Would definitely take a look at a PR.

**Comment by particlebanana** [comment]
Might have something to do with using the `.save` method here. Can you give a plan jane ES5 example here?

**Comment by tjwebb** [comment]
>    • `undefined` should be a binary (instead of "[object Object]", which is a object)

Agree with @devinivy, this is the classic symptom of using `_.deepClone` on a `Buffer`

**Comment by AnvarPK** [comment]
i am experiencing the same issue in sailsjs app, from client side(Ember) i am sending buffer to server, but at server side(sails)  showing an error

** ember **

  fileLoaded: function (file) {
            // console.log(file);
            // readAs="readAsFile"
            // console.log(file.name, file.type, file.size);
            
            // readAs="readAsArrayBuffer|readAsBinaryString|readAsDataURL|readAsText"
            // console.log(file.filename, file.type, file.data, file.size);
            console.log('fileLoded');
            Ember.$('.image img').attr('src', file.data);
            var proImage = this.get('store').createRecord('pro-image', {
                proImage: file
            });
            proImage.save().then(function (res) {
                // Success callback
                console.log(res.proImag);
            }, function (res) {
                // Error callback
                console.log(res);
            });

        }


** sails **
*** model***

// Pro-image.js

module.exports = {
  schema: true,
  attributes: {
    proImage:{
      type:'binary'
    }

  }
};


*** error ***

Invalid attributes sent to ProImage:
 • proImage
   • `undefined` should be a binary (instead of "[object Object]", which is a object)

**Comment by sailsbot** [comment]
Thanks for posting, @mattcodez.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by atiertant** [comment]
look at https://github.com/balderdashy/waterline/issues/1284
